WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,490 @@
+                boolean isEmpty = mSearchQuery == null || mSearchQuery.length() < MIN_SEARCH_QUERY_SIZE;
+                if (isEmpty) {
+                    String message = getString(R.string.stock_media_picker_initial_empty_text);
+                    String subMessage = getString(R.string.stock_media_picker_initial_empty_subtext);
+                    String link = "<a href='https://pexels.com/'>Pexels</a>";
+                    String html = message
+                                  + "<br /><br />"
+                                  + "<small>" + String.format(subMessage, link) + "</small>";
+                    txtEmpty.setMovementMethod(WPLinkMovementMethod.getInstance());
+                    txtEmpty.setText(Html.fromHtml(html));
+                } else {
+                    txtEmpty.setText(R.string.stock_media_picker_empty_results);
+                }
+            }
+        }
+    }
+
+    private void showProgress(boolean show) {
+        if (!isFinishing()) {
+            findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void submitSearch(@Nullable final String query, boolean delayed) {
+        mSearchQuery = query;
+
+        if (query == null || query.length() < MIN_SEARCH_QUERY_SIZE) {
+            mAdapter.clear();
+            showEmptyView(true);
+            return;
+        }
+
+        if (delayed) {
+            mHandler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    if (StringUtils.equals(query, mSearchQuery)) {
+                        submitSearch(query, false);
+                    }
+                }
+            }, 500);
+        } else {
+            requestStockMedia(query, 1);
+        }
+    }
+
+    private void requestStockMedia(@Nullable String searchTerm, int page) {
+        if (!NetworkUtils.checkConnection(this)) return;
+
+        if (page == 1) {
+            mAdapter.clear();
+        }
+
+        showProgress(true);
+        mViewModel.fetchStockMedia(searchTerm, page);
+    }
+
+    private void showSelectionBar() {
+        if (mSelectionBar.getVisibility() != View.VISIBLE) {
+            AniUtils.animateBottomBar(mSelectionBar, true);
+        }
+    }
+
+    private void hideSelectionBar() {
+        if (mSelectionBar.getVisibility() == View.VISIBLE) {
+            AniUtils.animateBottomBar(mSelectionBar, false);
+        }
+    }
+
+    private void notifySelectionCountChanged() {
+        int numSelected = mAdapter.getSelectionCount();
+        if (numSelected > 0) {
+            String label = String.format(getString(R.string.insert_count), numSelected);
+            mTextInsert.setText(label);
+            showSelectionBar();
+            if (numSelected == 1) {
+                ActivityUtils.hideKeyboardForced(mSearchView);
+            }
+        } else {
+            hideSelectionBar();
+        }
+    }
+
+    private void uploadSelection() {
```

Comment:
This is not yet implemented, but when it does, I think it should be in the view model.